components/collector/src/source_collectors/dependency_track/base.py

```diff
@@ -44,8 +44,8 @@ async def _get_source_responses(self, *urls: URL) -> SourceResponses:
         responses = SourceResponses()
         for url in urls:
             page_nr = 1  # Page numbers start at 1
-            total_count = self.PAGE_SIZE + 1  # Total count is still unknown, make it so big we get at least one page
-            while page_nr * self.PAGE_SIZE < total_count:
+            total_count = 1  # Total count is still unknown, make sure we retrieve at least one page
+            while (page_nr - 1) * self.PAGE_SIZE < total_count:
                 offsetted_url = URL(f"{url}{"&" if "?" in url else "?"}pageSize={self.PAGE_SIZE}&pageNumber={page_nr}")
```

Comment:
This url string munging is a bit dirty, we should have a helper method to merge http GET params from the query.
There are a lot of helpers provided by `urllib.parse`, avoiding to worry about a `#` fragment in the url.

Reply:
I don't see a helper in `urllib.parse` that lets one add one query argument to a URL. But I agree a helper is useful. I'll add one.